pkgs/http/lib/src/abortable.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'base_request.dart';
+import 'client.dart';
+import 'exception.dart';
+import 'streamed_response.dart';
+
+/// An HTTP request that can be aborted before it completes.
+abstract mixin class Abortable implements BaseRequest {
+  /// Completion of this future aborts this request (if the client supports
+  /// abortion).
+  ///
+  /// Requests/responses may be aborted at any time during their lifecycle.
+  ///
+  ///  * If completed before the request has been finalized and sent,
+  ///    [Client.send] completes with an [AbortedRequest] exception.
+  ///  * If completed after the response headers are available, or whilst
+  ///    streaming the response, clients inject [AbortedRequest] into the
+  ///    [StreamedResponse.stream] then close the stream.
+  ///  * If completed after the response is fully complete, there is no effect.
+  ///
+  /// A common pattern is aborting a request when another event occurs (such as
+  /// a user action): use a [Completer] to implement this. To implement a
+  /// timeout (to abort the request after a set time has elapsed), use
+  /// [Future.delayed].
+  ///
+  /// This future must not complete with an error.
+  ///
+  /// Some clients may not support abortion, or may not support this trigger.
+  abstract final Future<void>? abortTrigger;
+}
+
+/// Thrown when an HTTP request is aborted.
+///
+/// This exception is triggered when [Abortable.abortTrigger] completes.
+/// property for more info.
+class AbortedRequest extends ClientException {
```

Comment:
Has there been discussion on naming here?
Generally we'd follow `AbortedRequestException` or similar naming, right?

See https://api.dart.dev/dart-core/Exception-class.html

Reply:
`AbortedRequest` sounds like a "flavor" or `Request`, which this certainly is NOT.

Reply:
There is not been a discussion on naming. +1 to changing the exception name.

Reply:
I can see the potential confusion that it looks like an object that holds information about an aborted request. But the reason I was against calling it an exception (and against extending from client exception initially), is because it doesn't meet the docs on Exception:

> ...information ... about a failure

See also TickerCanceled: https://api.flutter.dev/flutter/scheduler/TickerCanceled-class.html

I'm not against changing the name to something like `RequestAborted` - but I wouldn't necessarily want to include "exception".